tools/cmake/cores/Cortex-M4.cmake

```diff
@@ -13,10 +13,10 @@ elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
         "-mfpu=none"
     )
     list(APPEND asm_compile_options
-        "-mcpu=Cortex-M4.no_fp"
+        "-mcpu=Cortex-M4"
     )
     list(APPEND link_options
-        "--cpu=Cortex-M4.no_fp"
+        "--cpu=Cortex-M4"
     )
```

Comment:
```suggestion
```

Then replace `c_cxx_compile_options ` with `common_options `.

Reply:
Hi all

I made the modifications.

After performing the following command:
 mbedtools -v compile -t ARM -m NUCLEO_WL55JC

I obtained the following error message:

"INFO: Found existing Mbed program at path 'C:\Users\reme\my-blinky'
INFO: This program does not contain an mbed_app.json config file.
Configuring project and generating build system...
-- The C compiler identification is ARMClang 6.13.100
-- The CXX compiler identification is ARMClang 6.13.100
-- The ASM compiler identification is ARMClang
-- Found assembler: C:/Keil_v5/ARM/ARMCLANG/bin/armclang.exe
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/Keil_v5/ARM/ARMCLANG/bin/armclang.exe
-- Check for working C compiler: C:/Keil_v5/ARM/ARMCLANG/bin/armclang.exe - broken
CMake Error at C:/Program Files/CMake/share/cmake-3.19/Modules/CMakeTestCCompiler.cmake:66 (message):
  The C compiler

    "C:/Keil_v5/ARM/ARMCLANG/bin/armclang.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/Users/reme/my-blinky/cmake_build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/PROGRA~1/CMake/bin/ninja.exe cmTC_39850 && [1/2] Building C object CMakeFiles\cmTC_39850.dir\testCCompiler.o
    [2/2] Linking C executable cmTC_39850.elf
    FAILED: cmTC_39850.elf 
    cmd.exe /C "cd . && C:\Keil_v5\ARM\ARMCLANG\bin\armlink.exe --cpu=cortex-m4 --map --cpu=Cortex-M.no_ftp --cpu=Cortex-M4   CMakeFiles\cmTC_39850.dir\testCCompiler.o -o cmTC_39850.elf  --list=cmTC_39850.elf.map && cd ."
    Fatal error: L3903U: Argument 'Cortex-M.no_ftp' not permitted for option 'cpu'.
     Finished: 0 information, 0 warning, 0 error and 1 fatal error messages.
    ninja: build stopped: subcommand failed."

I am able to remove the error message and thus to obtain a ".bin" file
by removing the two following lines  
- "--cpu=Cortex-M4.no_fp"
- "-mcpu=Cortex-M4.no_fp"

It is the reason why I am suggesting the modification.

Best regards

Reply:
I cannot reproduce the error locally as we don't seem to have any Cortex-M4 Mbed targets currently supported (I am assuming you trying to add support for NUCLEO_WL55JC?).

However, what I did try was attempting to build a K64F as a Cortex-M4 based target by modifying the `core` attribute in `targets.json`. It built successfully, I am therefore unable to comment about the issue you are observing as I do not have a full picture of what you are working with.

Is it possible that you may not have the correct shebang in your scatter file?

Reply:
Side note: The error says `Argument 'Cortex-M.no_ftp'` which is not `'Cortex-M.no_ftp'`...

Reply:
> Is it possible that you may not have the correct shebang in your scatter file?

Good catch !

Reply:
@hugueskamba I am assuming you trying to add support for NUCLEO_WL55JC?
Yes

Reply:
@hugueskamba  However, what I did try was attempting to build a K64F as a Cortex-M4 based target by modifying the core attribute in targets.json. It built successfully

Do you use mbedtools ?

Reply:
Yes. I used `mbedtools` (which subsequently calls CMake). I had to to make sure that the MCU core CMake module gets used in the configure stage. I added a `message()` in the `Cortex-M4.cmake` to make sure it was used.